core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java

```diff
@@ -751,6 +759,22 @@ public void testCase()
                         Optional.of(new LongLiteral("3"))));
     }
 
+    @Test
+    public void testSearchedCase()
```

Comment:
This seems unrelated to MERGE, but good to add. Make this a separate commit

Reply:
Done.